proxies/live/apiproxy/targets/apim.xml

```diff
@@ -42,6 +42,27 @@
         <Step>
           <Name>AssignMessage.AddIssuerHeader</Name>
         </Step>
+        <!-- If we require an ASID, fail if app.asid is null -->
+        {% if REQUIRE_ASID == 'True' %}
+        <Step>
+          <Name>RaiseFault.400BadRequest</Name>
+          <Condition>(app.asid is null)</Condition>
+        </Step>
+        {% endif %}
+        <!-- Populate the ASID var from the app -->
+        <Step>
+          <Name>AssignMessage.PopulateAsidFromApp</Name>
+          <Condition>(app.asid isNot null)</Condition>
+        </Step>
+        <Step>
+          <!-- Populate the ASID var with the default ASID -->
+          <Name>KeyValueMapOperations.GetDefaultAsid</Name>
+          <Condition>(app.asid is null)</Condition>
+        </Step>
+        <Step>
+          <!-- Add the ASID var as a header -->
+          <Name>AssignMessage.AddAsidHeader</Name>
+        </Step>
```

Comment:
Struggled to follow this initially, so the logic is: fail if there is no FROM_ASID in the apigee app in production, for other environments try to get a FROM_ASID from apigee app but fallback to a default value in an apigee keyvaluemap which is the generic platform FROM_ASID which has been updated to allow the required EPS message sets